src/app/podcasts/[id]/page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+import axios from "axios"
+import { useEffect, useState } from "react"
+import { usePathname, useSearchParams } from "next/navigation"
+import { useRouter } from "next/navigation"
+import Image from "next/image"
+import SpotifyLogo from "/public/images/logos/spotify-logo.svg"
+
+import { PodcastEpisode } from "../../../../utils/static/fetchPodcastsFromSpotify"
+
+const EpisodePage = () => {
+  const router = useRouter()
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+
+  const idSegment = pathname.split("/").pop()
+  const id = idSegment || searchParams.get("id") || ""
+
+  const [episode, setEpisode] = useState<PodcastEpisode | null>(null)
+
+  useEffect(() => {
+    const fetchEpisode = async () => {
+      try {
+        if (id) {
+          const response = await axios.get(
+            `https://spotify23.p.rapidapi.com/episode/?id=${id}`,
+            {
+              headers: {
+                "X-RapidAPI-Key":
+                  "ac41cab2aamsh587d2717936bec9p185b22jsna20c36b1f5ef",
+                "X-RapidAPI-Host": "spotify23.p.rapidapi.com",
+              },
+            }
+          )
+          console.log("Condition:", response.data.data.episodeUnionV2)
+          if (response.data?.data?.episodeUnionV2) {
+            const episodeData: PodcastEpisode =
+              response.data.data.episodeUnionV2
+            setEpisode(episodeData)
+          } else {
+            console.error("No video data found for the provided ID:", id)
+          }
+        } else {
+          console.warn("No video ID provided.")
+        }
+      } catch (error) {
+        console.error("Error fetching video data:", error)
+      }
+    }
+    fetchEpisode()
+  }, [id])
+  if (!episode) {
+    return <p>Loading episode data...</p>
+  }
+  const totalMilliseconds = episode.duration.totalMilliseconds
+  const minutes = Math.floor(totalMilliseconds / 60000)
+  const seconds = ((totalMilliseconds % 60000) / 1000).toFixed(0)
+
+  const publicationDate = episode.releaseDate.isoString
+  const formattedPublicationDate = publicationDate.split("T")[0]
```

Comment:
We typically do date formating in date utility class. Here is one example: 

**formatDate.ts**

```
import { parseISO, parseJSON, format, isAfter, formatDistanceToNow } from 'date-fns';
import { enGB } from 'date-fns/locale';

export enum DateFormats {
  fullNumericalDate = 'dd.MM.yyyy.', // e.g. 08.11.2021.
  hoursAndMinutes = 'HH:mm', // e.g. 01:35
  fullNumericalDateWithTime = 'dd.MM.yyyy hh:mm',
  fullDateWithMonthName = 'LLLL d, yyyy', // e.g. November 8, 2021
}

const formatDate = (date: string, formatType: string): string => format(parseISO(date), formatType);

export const isPostedAfterLastLogin = (date, dateToCompare): boolean => isAfter(parseJSON(date), dateToCompare);

export const checkTimeDistance = updatedAt => {
  let timeAgo = '';
  if (updatedAt) {
    const date = parseISO(updatedAt);
    const timePeriod = formatDistanceToNow(date, { locale: enGB });

    timeAgo = `${timePeriod} ago`;
  }

  return timeAgo;
};

export default formatDate;

```